s3/backend.go

```diff
@@ -334,25 +336,20 @@ func (s *s3) DeleteObject(ctx context.Context, bucketName, objectName string) (g
 //
 // The size can be used if the backend needs to read the whole reader; use
 // gofakes3.ReadAll() for this job rather than ioutil.ReadAll().
-// TODO: Metadata is currently ignored. The backend requires an update to
-// support it.
 func (s *s3) PutObject(ctx context.Context, bucketName, key string, meta map[string]string, input io.Reader, size int64) (gofakes3.PutObjectResult, error) {
-	opts := api.UploadObjectOptions{
+	if ur, err := s.w.UploadObject(ctx, input, bucketName, key, api.UploadObjectOptions{
 		ContentLength: size,
-	}
-	if ct, ok := meta["Content-Type"]; ok {
-		opts.MimeType = ct
-	}
-	ur, err := s.w.UploadObject(ctx, input, bucketName, key, opts)
-	if err != nil && strings.Contains(err.Error(), api.ErrBucketNotFound.Error()) {
+		Metadata:      meta,
+	}); err != nil && strings.Contains(err.Error(), api.ErrBucketNotFound.Error()) {
 		return gofakes3.PutObjectResult{}, gofakes3.BucketNotFound(bucketName)
 	} else if err != nil {
 		return gofakes3.PutObjectResult{}, gofakes3.ErrorMessage(gofakes3.ErrInternal, err.Error())
+	} else {
```

Comment:
I'd move this out of the `else`. I'm surprised neither vet or lint complained about this since it usually does for me.

Reply:
I was trying to keep it consistent with all `s3` methods in `backend.go`. I reverted it though but then else updated all other methods so we're consistent throughout the file.